processor/logdedupprocessor/counter.go

```diff
@@ -144,21 +162,32 @@ func (a *logCounter) Increment() {
 	a.count++
 }
 
-// getLogKey creates a unique hash for the log record to use as a map key
-/* #nosec G104 -- According to Hash interface write can never return an error */
-func getLogKey(logRecord plog.LogRecord) [8]byte {
-	hasher := fnv.New64()
-	attrHash := pdatautil.MapHash(logRecord.Attributes())
+// getResourceKey creates a unique hash for the resource to use as a map key
+func getResourceKey(resource pcommon.Resource) uint64 {
+	hasher := xxhash.New()
+	attrsHash := pdatautil.MapHash(resource.Attributes())
+	_, _ = hasher.Write(attrsHash[:])
+	return hasher.Sum64()
+}
 
-	hasher.Write(attrHash[:])
+// getScopeKey creates a unique hash for the scope to use as a map key
+func getScopeKey(scope pcommon.InstrumentationScope) uint64 {
```

Comment:
Can we add this to pdatautil?

Reply:
Extended pdatautil to do more of the work here instead in [a716f65](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34607/commits/a716f655000a93824213245b7f2c3da440105719)